src/rules/attr-whitespace.js

```diff
@@ -0,0 +1,44 @@
+HTMLHint.addRule({
+  id: 'attr-whitespace',
+  description:
+    'All attributes should be separated by only one space and not have leading/trailing whitespace.',
+  init: function(parser, reporter, options) {
+    var self = this;
+    var exceptions = Array.isArray(options) ? options : [];
+    //console.log(exceptions);
+    if (exceptions.length == 0) {
```

Comment:
In JS it is recommended to use triple equals (===) instead of just two equals  (==).